### DISCUSSION TITLE
Create a Docker image

### DISCUSSION THREAD

**Discussion by adrianpaniagualeon** [mgilangjanuar/teledrive#217] [Ideas]
It would be great to have a Docker image to speed up the installation process more.

Thanks

**Comment by mgilangjanuar** [comment]
Yes, we're on it to create a docker image for making the installation process easier 😆

**Comment by mgilangjanuar** [comment]
Update: it's done using docker-compose 🎊

**Comment by WolfRamiO74** [comment]
hi there, the docker installation methos fail... i test it on ubuntu 22.10 and debian 11

**Reply by birdup000** [reply]
> hi there, the docker installation methos fail... i test it on ubuntu 22.10 and debian 11

Hi are you reviewing error about data or migrations?

**Reply by WolfRamiO74** [reply]
**docker compose up -d command**

root@UnuntuTEST:~/teledrive/docker# docker compose up -d
WARN[0000] The "sad" variable is not set. Defaulting to a blank string. 
WARN[0000] The "sad" variable is not set. Defaulting to a blank string. 
WARN[0000] The "sad" variable is not set. Defaulting to a blank string. 
WARN[0000] The "sad" variable is not set. Defaulting to a blank string. 
[+] Building 2.8s (6/18)                                                                                                                 
 => [internal] load build definition from Dockerfile                                                                                0.2s
 => => transferring dockerfile: 538B                                                                                                0.0s
 => [internal] load .dockerignore                                                                                                   0.3s
 => => transferring context: 2B                                                                                                     0.0s
 => [internal] load metadata for docker.io/library/node:18.16.0                                                                     1.2s
 => CACHED [ 1/14] FROM docker.io/library/node:18.16.0@sha256:671ee8d49ce2a691fc3082203c5deb9522e0c80042aa0ff40c07f4a25e63668a      0.0s
 => [internal] load build context                                                                                                   0.2s
 => => transferring context: 40.25kB                                                                                                0.0s
 => ERROR [ 2/14] RUN echo Fix error: `cd docker && chmod -R 777 data'                                                              1.3s
------                                                                                                                                   
 > [ 2/14] RUN echo Fix error: `cd docker && chmod -R 777 data':
#0 1.064 /bin/sh: 1: Syntax error: EOF in backquote substitution
------
failed to solve: process "/bin/sh -c echo Fix error: `cd docker && chmod -R 777 data'" did not complete successfully: exit code: 2
root@UnuntuTEST:~/teledrive/docker# 


**./install.docker.new.sh**

root@UnuntuTEST:~/teledrive# ./install.docker.new.sh
Node Version: v18.7.0
cURL Version: curl 7.85.0 (x86_64-pc-linux-gnu) libcurl/7.85.0 OpenSSL/3.0.5 zlib/1.2.11 brotli/1.0.9 zstd/1.5.2 libidn2/2.3.3 libpsl/0.21.0 (+libidn2/2.3.2) libssh/0.9.6/openssl/zlib nghttp2/1.49.0 librtmp/2.3
Docker Version: Docker version 23.0.4, build f480fb1
Docker Compose Version: Docker Compose version v2.17.2
From https://github.com/mgilangjanuar/teledrive
 * branch            main       -> FETCH_HEAD
Already up to date.
./install.docker.new.sh: line 70: export: `#': not a valid identifier


**./install.docker.sh**

root@UnuntuTEST:~/teledrive# ./install.docker.sh
Node Version: v18.7.0
cURL Version: curl 7.85.0 (x86_64-pc-linux-gnu) libcurl/7.85.0 OpenSSL/3.0.5 zlib/1.2.11 brotli/1.0.9 zstd/1.5.2 libidn2/2.3.3 libpsl/0.21.0 (+libidn2/2.3.2) libssh/0.9.6/openssl/zlib nghttp2/1.49.0 librtmp/2.3
Docker Version: Docker version 23.0.4, build f480fb1
Docker Compose Version: docker-compose version 1.29.2, build unknown
HEAD is now at 4ef9ba3 Instructions add
From https://github.com/mgilangjanuar/teledrive
 * branch            main       -> FETCH_HEAD
Already up to date.
./install.docker.sh: line 45: export: `#': not a valid identifier
root@UnuntuTEST:~/teledrive#

**Reply by WolfRamiO74** [reply]
> > hi there, the docker installation methos fail... i test it on ubuntu 22.10 and debian 11
> 
> Hi are you reviewing error about data or migrations?
![image](https://user-images.githubusercontent.com/12602710/234221497-f052dcd8-e1d8-43d6-9617-8297cc0005b9.png)
